lib/mayaUsd/undo/UsdUndoStateDelegate.cpp

```diff
@@ -0,0 +1,450 @@
+    const SdfDataRefPtr& deletedData)
+{
+    _setMessageAlreadyShowed = true;
+
+    TF_DEBUG(USDMAYA_UNDOSTATEDELEGATE).Msg("Inverting deleting spec at '%s'\n", path.GetText());
+
+    CreateSpec(path, deletedSpecType, inert);
+}
+
+void UsdUndoStateDelegate::invertMoveSpec(const SdfPath& oldPath, const SdfPath& newPath)
+{
+    _setMessageAlreadyShowed = true;
+
+    TF_DEBUG(USDMAYA_UNDOSTATEDELEGATE).Msg("Inverting move of '%s' to '%s'\n", oldPath.GetText(), newPath.GetText());
+
+    MoveSpec(newPath, oldPath);
+}
+
+void UsdUndoStateDelegate::invertPushTokenChild(
+    const SdfPath& parentPath,
+    const TfToken& fieldName,
+    const TfToken& value)
+{
+    _setMessageAlreadyShowed = true;
+
+    TF_DEBUG(USDMAYA_UNDOSTATEDELEGATE)
+        .Msg("Inverting push field '%s' of '%s'\n", fieldName.GetText(), parentPath.GetText());
+
+    PopChild(parentPath, fieldName, value);
+}
+
+void UsdUndoStateDelegate::invertPushPathChild(
+    const SdfPath& parentPath,
+    const TfToken& fieldName,
+    const SdfPath& value)
+{
+    _setMessageAlreadyShowed = true;
+
+    TF_DEBUG(USDMAYA_UNDOSTATEDELEGATE)
+        .Msg("Inverting push field '%s' of '%s'\n", fieldName.GetText(), parentPath.GetText());
+
+    PopChild(parentPath, fieldName, value);
+}
+
+void UsdUndoStateDelegate::invertPopTokenChild(
+    const SdfPath& parentPath,
+    const TfToken& fieldName,
+    const TfToken& value)
+{
+    _setMessageAlreadyShowed = true;
+
+    TF_DEBUG(USDMAYA_UNDOSTATEDELEGATE)
+        .Msg("Inverting pop field '%s' of '%s'\n", fieldName.GetText(), parentPath.GetText());
+
+    PushChild(parentPath, fieldName, value);
+}
+
+void UsdUndoStateDelegate::invertPopPathChild(const SdfPath& parentPath, const TfToken& fieldName, const SdfPath& value)
+{
+    _setMessageAlreadyShowed = true;
+
+    TF_DEBUG(USDMAYA_UNDOSTATEDELEGATE)
+        .Msg("Inverting pop field '%s' of '%s'\n", fieldName.GetText(), parentPath.GetText());
+
+    PushChild(parentPath, fieldName, value);
+}
+
+void UsdUndoStateDelegate::invertSetFieldDictValueByKey(const SdfPath& path, const TfToken& fieldName, const TfToken& keyPath, const VtValue& inverse)
+{
+    _setMessageAlreadyShowed = true;
+
+    TF_DEBUG(USDMAYA_UNDOSTATEDELEGATE)
+            .Msg("Inverting Field '%s' By Key '%s' for Spec '%s'\n", fieldName.GetText(), keyPath.GetText(), path.GetText());
+
+    SetFieldDictValueByKey(path, fieldName, keyPath, inverse);
+}
+
+void UsdUndoStateDelegate::invertSetTimeSample(const SdfPath& path, double time, const VtValue& inverse)
+{
+    _setMessageAlreadyShowed = true;
+
+    TF_DEBUG(USDMAYA_UNDOSTATEDELEGATE)
+        .Msg("Inverting TimeSample '%f' for Spec '%s'\n", time, path.GetText());
+
+    SetTimeSample(path, time, inverse);
+}
+
+void UsdUndoStateDelegate::_OnSetField(const SdfPath& path, const TfToken& fieldName, const VtValue& value)
+{
+    _MarkCurrentStateAsDirty();
+
+    // early return if we are not insdide an UsdUndoBlock
```

Comment:
"insdide" -->"inside" and elsewhere.

Reply:
addressed in c51974c2ad2d08584837f439f99474d210b67d29